aaelf32/aaelf32.rst

```diff
@@ -0,0 +1,3289 @@
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+R_ARM_THM_BF16, R_ARM_THM_BF12 and R_ARM_THM_BF18 are used to relocate branch
+targets for the Branch Future instructions in Armv8.1-M Mainline. The formation
+of the initial addend in a REL type relocation for the various instruction
+classes is described in `Armv8.1-M Mainline Branch Future relocation actions by
+instruction type`_. The immediates mentioned in this table are the ones used to
+represent the offset from the targetted branch instruction to the branch target
+using a two's complement immediate. That is this immediate should be the same
+as the one being used by the targetted branch instruction. Insn modification
+describes how the result X is written back to the instruction; Result_Mask is
+the value of X after the masking operation described in `Armv8.1-M Mainline
+Branch Future relocations table`_ has been applied.
+
+In all cases the following semantics apply:
+
+* A Branch Future to an out of range label must either be replaced with a NOP
+  or re-use an existing trampoline that jumps to the relocation target, it must
+  not create a new trampoline. (It is up to the implementation to decide which
+  of these options is preferable).
+* No errors are required.
+
+.. _Armv8.1-M Mainline Branch Future relocations table:
+.. table:: Armv8.1-M Mainline Branch Future relocations
+
+   +------+---------------------------------+----------+----------------------------------------+--------------------+
+   | Code | Name                            | Overflow | Instruction                            | Result Mask        |
+   +======+=================================+==========+========================================+====================+
+   | 136  | :code:`R_ARM_THM_BF16`          | Replace  | BF 16-bit immediate                    | X & 0x0001FFFE     |
+   |      |                                 | with NOP |                                        |                    |
+   +------+---------------------------------+----------+----------------------------------------+--------------------+
+   | 137  | :code:`R_ARM_THM_BF12`          | Replace  | BFCSEL 12-bit immediate                | X & 0x00001FFE     |
+   |      |                                 | with NOP |                                        |                    |
+   +------+---------------------------------+----------+----------------------------------------+--------------------+
+   | 138  | :code:`R_ARM_THM_BF18`          | Replace  | BFL 18-bit immediate                   | X & 0x0007FFFE     |
+   |      |                                 | with NOP |                                        |                    |
+   +------+---------------------------------+----------+----------------------------------------+--------------------+
+
+.. _Armv8.1-M Mainline Branch Future relocation actions by instruction type:
+
+.. table:: Armv8.1-M Mainline Branch Future relocation actions by instruction type
+  :widths: 15 40 45
```

Comment:
as described above on the version history table, please use a class to specify the width